client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -194,23 +197,20 @@ const CheckoutThankYou = React.createClass( {
 	},
 
 	render() {
-		let purchases = null,
-			failedPurchases = null,
+		let purchases = [],
+			failedPurchases = [],
 			wasJetpackPlanPurchased = false,
 			wasDotcomPlanPurchased = false;
+
 		if ( this.isDataLoaded() && ! this.isGenericReceipt() ) {
 			purchases = getPurchases( this.props );
 			failedPurchases = getFailedPurchases( this.props );
 			wasJetpackPlanPurchased = purchases.some( isJetpackPlan );
 			wasDotcomPlanPurchased = purchases.some( isDotComPlan );
 		}
 
-		const userCreatedMoment = moment( this.props.userDate ),
-			isNewUser = userCreatedMoment.isAfter( moment().subtract( 2, 'hours' ) ),
-			goBackText = this.props.selectedSite ? this.translate( 'Back to my site' ) : this.translate( 'Register Domain' );
-
 		// this placeholder is using just wp logo here because two possible states do not share a common layout
-		if ( ! purchases && ! failedPurchases && ! this.isGenericReceipt() ) {
+		if ( ! purchases.length && ! failedPurchases.length && ! this.isGenericReceipt() ) {
```

Comment:
That's personal preference but I like using [`isEmpty()`](https://lodash.com/docs#isEmpty) in those cases since it makes the code reads like a sentence.